crates/services/src/auth/near.rs

```diff
@@ -0,0 +1,401 @@
+use chrono::{DateTime, Duration, Utc};
+use config::NearConfig;
+use near_api::{signer::NEP413Payload, types::Signature, AccountId, NetworkConfig, PublicKey};
+use std::fmt::{Display, Formatter};
+use std::sync::Arc;
+use url::Url;
+
+use super::ports::{AuthServiceTrait, NearNonceRepository, OAuthUserInfo, Session};
+
+const MAX_NONCE_AGE_MS: u64 = 5 * 60 * 1000; // 5 minutes
+const EXPECTED_MESSAGE: &str = "Sign in to NEAR AI Cloud";
+
+/// Custom error type for NEAR authentication
+#[derive(Debug)]
```

Comment:
We should use thiserror. Example:
https://github.com/nearai/cloud-api/blob/3e7f179bd99029dc0285f3b62227bd02c6cc29fc/crates/services/src/auth/ports.rs#L102-L138

Reply:
updated